compiler/src/main/java/io/neow3j/compiler/NeoMethod.java

```diff
@@ -465,13 +465,66 @@ public NeoVariable getParameterByJVMIndex(int index) {
      */
     public void convert(CompilationUnit compUnit) throws IOException {
         AbstractInsnNode insn = asmMethod.instructions.get(0);
+        throwIfMethodHasSDKRelatedReturnOrArgType(asmMethod);
         while (insn != null) {
             insn = Compiler.handleInsn(insn, this, compUnit);
             insn = insn.getNext();
         }
         insertTryCatchBlocks();
     }
 
+    private void throwIfMethodHasSDKRelatedReturnOrArgType(MethodNode asmMethod) {
+        Type returnType = Type.getReturnType(asmMethod.desc);
+        if (isSDKRelatedNeow3jType(returnType.getClassName())) {
+            throw new CompilerException(
+                    format("The neow3j compiler does not support SDK-related types. Type '%s' used as return type of " +
+                            "method '%s' is not supported.", returnType.getClassName(), asmMethod.name));
+        }
+        for (Type argType : Type.getArgumentTypes(asmMethod.desc)) {
+            if (isSDKRelatedNeow3jType(argType.getClassName())) {
+                throw new CompilerException(
+                        format("The neow3j compiler does not support SDK-related types. Type '%s' used for an " +
+                                "argument of method '%s' is not supported.", argType.getClassName(), asmMethod.name));
+            }
+        }
+    }
+
+    public static void throwIfSDKRelatedType(String classOrFullyQualifiedName) {
+        if (isSDKRelatedNeow3jType(classOrFullyQualifiedName)) {
+            throw new CompilerException(
+                    format("The neow3j compiler does not support SDK-related types. Type '%s' is not supported.",
+                            classOrFullyQualifiedName));
+        }
+    }
+
+    private static boolean isSDKRelatedNeow3jType(String classOrFullyQualifiedName) {
+        return isSDKRelatedNeow3jType(classOrFullyQualifiedName, "\\.");
+    }
+
+    // Returns true, if the provided name starts with 'io' and 'neow3j' and is SDK-related.
+    public static boolean isSDKRelatedNeow3jType(String classOrFullyQualifiedName, String regex) {
```

Comment:
Why are you exposing the regex outside of this method? For me, it doesn't make sense. The regex is specific for this method implementation, and the regex will not change.

Reply:
Moreover, it's not a regex. It's just a separator, right?

Reply:
You're right, I changed the code to handle it in a more consistent way.